src/sortStudents.ts

```diff
@@ -1,16 +1,50 @@
 
 export interface Student {
-  // describe Student interface
+  name: string;
+  surname: string;
+  age: number;
+  married: boolean;
+  grades: number[];
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  Grades = 'grades',
 }
 
-// create SortOrder type
-export type SortOrder;
+type SortOrder = 'asc' | 'desc';
 
+export function sortStudents(
+  students: Student[], sortBy: SortType, order: SortOrder,
+):[string] {
+  const learner = [...students];
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+  function getAverage(grades:number[]):number {
+    return grades.reduce((prev, element) => prev + element) / grades.length;
+  }
+
+  switch (sortBy) {
+    case SortType.Name:
+    case SortType.Surname:
+      return order === 'asc'
+        ? learner.sort((a, b) => a[sortBy].localeCompare(b[sortBy]))
+        : learner.sort((a, b) => b[sortBy].localeCompare(a[sortBy]));
+
+    case SortType.Age:
+    case SortType.Married:
+      return order === 'asc'
+        ? learner.sort((a, b) => +a[sortBy] - +b[sortBy])
+        : learner.sort((a, b) => +b[sortBy] - +a[sortBy]);
```

Comment:
better don't use the `+` shorthand - but a Number casting instead
```suggestion
        ? learner.sort((a, b) => Number(a[sortBy]) - Number(b[sortBy]))
        : learner.sort((a, b) => Number(b[sortBy]) - Number(a[sortBy]));
```